test/call.c

```diff
@@ -884,6 +898,104 @@ int test_call_video(void)
 }
 
 
+int test_call_change_videodir(void)
+{
+	struct fixture fix, *f = &fix;
+	struct vidisp *vidisp = NULL;
+	enum sdp_dir a_video_ldir, a_video_rdir, b_video_ldir, b_video_rdir;
+	int err = 0;
+
+	conf_config()->video.fps = 100;
+	conf_config()->video.enc_fmt = VID_FMT_YUV420P;
+
+	fixture_init(f);
+
+	/* to enable video, we need one vidsrc and vidcodec */
+	mock_vidcodec_register();
+
+	err = mock_vidisp_register(&vidisp, mock_vidisp_handler, f);
+	TEST_ERR(err);
+
+	err = module_load(".", "fakevideo");
+	TEST_ERR(err);
+
+	f->behaviour = BEHAVIOUR_ANSWER;
+	f->estab_action = ACTION_NOTHING;
+	f->accept_session_updates = true;
+
+	/* Make a call from A to B */
+	err = ua_connect(f->a.ua, 0, NULL, f->buri, VIDMODE_ON);
+	TEST_ERR(err);
+
+	/* run main-loop with timeout, wait for events */
+	err = re_main_timeout(10000);
+	TEST_ERR(err);
+	TEST_ERR(fix.err);
+
+	/* verify that video was enabled for this call */
+	ASSERT_EQ(1, fix.a.n_established);
+	ASSERT_EQ(1, fix.b.n_established);
+
+	ASSERT_TRUE(call_has_video(ua_call(f->a.ua)));
+	ASSERT_TRUE(call_has_video(ua_call(f->b.ua)));
+
+	/* Set video inactive */
+	err = call_set_video_dir(ua_call(f->a.ua), SDP_INACTIVE);
+	TEST_ERR(err);
+	err = re_main_timeout(10000);
+	TEST_ERR(err);
+	ASSERT_TRUE(call_has_video(ua_call(f->a.ua)));
+	ASSERT_TRUE(call_has_video(ua_call(f->b.ua)));
+
+	a_video_ldir = sdp_media_ldir(stream_sdpmedia(
+			video_strm(call_video(ua_call(f->a.ua)))));
+	a_video_rdir = sdp_media_rdir(stream_sdpmedia(
+			video_strm(call_video(ua_call(f->a.ua)))));
+	b_video_ldir = sdp_media_ldir(stream_sdpmedia(
+			video_strm(call_video(ua_call(f->b.ua)))));
+	b_video_rdir = sdp_media_rdir(stream_sdpmedia(
+			video_strm(call_video(ua_call(f->b.ua)))));
+
+	ASSERT_TRUE(a_video_ldir == SDP_INACTIVE);
+	ASSERT_TRUE(a_video_rdir == SDP_SENDRECV);
+	ASSERT_TRUE(b_video_ldir == SDP_SENDRECV);
+	ASSERT_TRUE(b_video_rdir == SDP_INACTIVE);
+
+	/* Set video sendrecv */
+	err = call_set_video_dir(ua_call(f->a.ua), SDP_SENDRECV);
+	TEST_ERR(err);
+	err = re_main_timeout(10000);
+	err |= re_main_timeout(10000);
+	err |= re_main_timeout(10000);
```

Comment:
The `if` in the `event_handler()` should be extended that these multiple `re_main_timeout()` become one.